app/src/main/java/com/vultisig/wallet/ui/models/cosmosstaking/CosmosDelegateViewModel.kt

```
@@ -0,0 +1,285 @@
+            onError = { e -> setError(e.message ?: "Failed to build delegate transaction") }
+        ) {
+            val coin = coin ?: return@safeLaunch setError("Wallet not loaded yet")
+            val vault =
+                withContext(Dispatchers.IO) { vaultRepository.get(route.vaultId) }
+                    ?: return@safeLaunch setError("Vault not found: ${route.vaultId}")
+
+            // Sanity-check at the boundary — the resolver runs the full bech32 preflight again
+            // before producing SignDoc bytes, but failing fast here gives the user a clearer error.
+            try {
+                ValidatorBech32Preflight.validate(validator.operatorAddress, coin.chain)
+            } catch (e: ValidatorBech32Preflight.ValidatorBech32Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)